src/ConnectedMode.UnitTests/ServerSentEvents/SSESessionManagerTests.cs

```diff
@@ -97,16 +136,31 @@ public void OnSolutionChanged_WhenChangesFromConnectedToConnected_CancelsSession
         sessionMock2.Verify(session => session.Dispose(), Times.Never);
     }
 
+    [TestMethod]
+    public void OnSolutionChanged_WhenAttemptsToConnectToAlreadyConnectedProject_KeepsPreviousSessionAlive()
```

Comment:
See my comment in the product code. I think this is dangerous.
The user can force a connection to restart by right-clicking in the team explorer and picking "Update". This will disable it.

Reply:
see my reply in that thread